### ISSUE TITLE
SC2032 and SC2033 for find -type f probably should not warn

### ISSUE THREAD

**Issue by focusaurus** [koalaman/shellcheck#1537]
#### For bugs
- Rule Id: SC2032, SC2033 
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6 (also online)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
alias f="fasd -f"
find . -type f
```

#### Here's what shellcheck currently says:

```
 
Line 2:
alias f="fasd -f"
      ^-- SC2032: Use own script or sh -c '..' to run this from find.
 
Line 3:
find . -type f
             ^-- SC2033: Shell functions can't be passed to external commands.
```

#### Here's what I wanted or expected to see:

I'm not sure of exactly how I think this rule might be tweaked, but before I saw this warning it had never even occurred to me that someone might think you could pass a reference to a shell alias like this. I guess it makes sense to me what the warning is getting at, but I think maybe the rule should focus on `find -exec` and be silent about `find -type`. Again I'm not totally sure what my suggestion is here but I think there might be a reasonable way to make this warning more accurate and narrow in scope.

**Comment by koalaman** [comment]
This was indeed annoying. The warning triggered for any unquoted word passed to a limited set of commands like `xargs`, `sudo` and `find`. It's not a common warning, but the false positive rate was really bad. In particular, it turns out there's quite a few functions called `install` that invoked commands like `sudo apt-get install ..`

It's now been tweaked to only look at specific arguments, such as the first non-flag for `xargs` and `sudo`, and for anything after `-exec` and similar for `find`. It's not perfect and kills some true positives, but the number of false positives should be down dramatically.
